tests/glsl-intrinsic/shader-subgroup/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,200 @@
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+//TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl
+//TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda 
+// not testing cpp due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+//#define TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
+//#define TEST_when_glsl_subgroupInclusiveAdd_is_not_bugged 
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2
+{
+    uint data[];
+} outputBuffer;
+
+#define local_size_x_v 32
+layout(local_size_x = local_size_x_v) in;
+
+__generic<T : __BuiltinLogicalType>
+bool test1Logical() {
+    return true
+#if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
+        && subgroupAnd(T(1)) == T(1)
+        && subgroupOr(T(1)) == T(1)
+        && subgroupXor(T(1)) == T(0) 
+        && subgroupInclusiveAnd(T(1)) == T(1)
+        && subgroupInclusiveOr(T(1)) == T(1)
+#ifdef TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
+        && subgroupInclusiveXor(T(1)) == T(0)
+#endif // #ifdef TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
```

Comment:
Or works but not Xor?
Aren't they almost same??

Reply:
SPIR-V and GLSL both produce different results; the code that runs either is also very simple, so it looks like a different error than the glsl.meta code.

the result should be 1+1+1+1 for thread 3, glsl returns 1 (wrong), spir-v returns 4 (correct). It's likely a bug outside our system, but that is significantly outside domain of what I am doing.